specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json

```diff
@@ -2116,6 +2462,184 @@
       "required": [
         "value"
       ]
+    },
+    "OSVersionProperties": {
+      "description": "OS version properties.",
+      "type": "object",
+      "properties": {
+        "family": {
+          "description": "The family of this OS version.",
+          "type": "string",
+          "readOnly": true
+        },
+        "familyLabel": {
+          "description": "The family label of this OS version.",
+          "type": "string",
+          "readOnly": true
+        },
+        "version": {
+          "description": "The OS version.",
+          "type": "string",
+          "readOnly": true
+        },
+        "label": {
+          "description": "The OS version label.",
+          "type": "string",
+          "readOnly": true
+        },
+        "isDefault": {
+          "description": "Specifies whether this is the default OS version for its family.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "isActive": {
+          "description": "Specifies whether this OS version is active.",
+          "type": "boolean",
+          "readOnly": true
+        }
+      }
+    },
+    "OSVersion": {
+      "description": "Describes a cloud service OS version.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Resource Id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
```

Comment:
Similar to other location resource types in compute, I guess locations/cloudServiceOsVersions resource also uses ProxyOnly, LocationBased routing. Property 'location' cannot be used as a top level property for proxy resources.
Refer the NOTE section in the RP documentation here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope

Reply:
I see that the note says that location is not applicable in the request body. However, these ProxyOnly resources are only available through GET. Additionally readonly: true is specified. So location would only be applied for the response body in this situation.

Reply:
As per offline discussion, as cloudServiceOsVersions are associated with a location, including location in response sounds reasonable.